File: app/src/main/java/com/imangazalievm/bubbble/domain/repository/CommentsRepository.java
package com.imangazalievm.bubbble.domain.repository;


import com.imangazalievm.bubbble.domain.models.Comment;
import com.imangazalievm.bubbble.domain.models.ShotCommentsRequestParams;

import java.util.List;

import io.reactivex.Single;

public interface CommentsRepository {

    Single<List<Comment>> getComments(ShotCommentsRequestParams requestParams);

}

File: app/src/main/java/com/imangazalievm/bubbble/data/shots/CommentsRepository.kt
package com.imangazalievm.bubbble.data.shots

import com.imangazalievm.bubbble.data.global.network.DribbbleApi
import com.imangazalievm.bubbble.domain.global.models.Comment
import com.imangazalievm.bubbble.domain.global.models.ShotCommentsRequestParams
import javax.inject.Inject

class CommentsRepository @Inject constructor(private val dribbbleApi: DribbbleApi) {

    suspend fun getComments(requestParams: ShotCommentsRequestParams): List<Comment> {
        return dribbbleApi.getShotComments(
            requestParams.shotId,
            requestParams.page,
            requestParams.pageSize
        )
    }

}